File: week7_test/distinct_lines.pl
#!/usr/bin/perl -w
use strict;
exit 1 if (!@ARGV);

my %dic=();
my $distinct = 0;
my $len = 0;
my $flag = 0;
while (my $line = <STDIN>){
	chomp $line;
	$line =~ s/ +/ /g;
	$line =~ s/ *//;
	#print "Debug: $line\n";
	$distinct++ if(! exists $dic{lc($line)});
	$dic{lc($line)} = 1;
	$len++;
	 if($ARGV[0] == $distinct){
                $flag = 1;
                last;
        }
}
if($flag){
	print "$distinct distinct lines seen after $len lines read.\n"
}else{
	print "End of input reached after $len lines read - $ARGV[0] different lines not seen.\n"
}

File: week9_test/distinct_lines.py
#!/usr/bin/python3

import sys,re

if len(sys.argv) < 2:
	sys.exit(1)

dic = {}
num = int(sys.argv[1])
count = 0
for line in sys.stdin.readlines():
	if len(dic) == num:
		break
	line=re.sub('\s+',"",line)
	dic[line.lower()]=1
	count += 1

if(len(dic) == num):
	print("%d distinct lines seen after %d lines read." % (num, count))
else:
	print("End of input reached after %d lines read -  %d different lines not seen." % (count,num))
